src/Analyzer.BicepProcessor/BicepTemplateProcessor.cs

```diff
@@ -45,36 +66,17 @@ public class BicepTemplateProcessor
         /// <returns>The compiled template as a <c>JSON</c> string and its source map.</returns>
         public static (string, BicepMetadata) ConvertBicepToJson(string bicepPath)
         {
-            var bicepPathUri = new Uri(bicepPath);
-            using var stringWriter = new StringWriter();
-
-            var configuration = configurationManager.GetConfiguration(bicepPathUri);
-
-            var featureProviderFactory = IFeatureProviderFactory.WithStaticFeatureProvider(
-                new SourceMapFeatureProvider(new FeatureProvider(configuration)));
-            var apiVersionProvider = new ApiVersionProvider(
-                featureProviderFactory.GetFeatureProvider(bicepPathUri), namespaceProvider);
-            var apiVersionProviderFactory = IApiVersionProviderFactory.WithStaticApiVersionProvider(apiVersionProvider);
-
-            var moduleDispatcher = new ModuleDispatcher(
-                new DefaultModuleRegistryProvider(
-                    fileResolver,
-                    new ContainerRegistryClientFactory(tokenCredentialFactory),
-                    new TemplateSpecRepositoryFactory(tokenCredentialFactory),
-                    featureProviderFactory,
-                    configurationManager),
-                configurationManager);
-            var workspace = new Workspace();
-            var sourceFileGrouping = SourceFileGroupingBuilder.Build(fileResolver, moduleDispatcher, workspace, PathHelper.FilePathToFileUrl(bicepPath));
-
-            // Pull modules optimistically
-            if (moduleDispatcher.RestoreModules(moduleDispatcher.GetValidModuleReferences(sourceFileGrouping.GetModulesToRestore())).Result)
+            if (BicepCompiler == null)
             {
-                // Modules had to be restored - recompile
-                sourceFileGrouping = SourceFileGroupingBuilder.Rebuild(moduleDispatcher, workspace, sourceFileGrouping);
+                var serviceCollection = new ServiceCollection();
+                serviceCollection.AddBicepCore();
+
+                var services = serviceCollection.BuildServiceProvider();
+                BicepCompiler = services.GetRequiredService<BicepCompiler>();
             }
 
-            var compilation = new Compilation(featureProviderFactory, namespaceProvider, sourceFileGrouping, configurationManager, apiVersionProviderFactory, new LinterAnalyzer());
+            using var stringWriter = new StringWriter();
+            var compilation = BicepCompiler.CreateCompilation(new Uri(bicepPath)).Result; // TODO: sync call OK?
```

Comment:
>// TODO: sync call OK? [](http://example.com/codeflow?start=90&length=22)

I mean, if the calling method is synchronous, why not? :)

Is there an opportunity for Template Analyzer to do other work asynchronously while the Bicep compiler is compiling?